compiler/erg_parser/parse.rs

```diff
@@ -905,14 +926,24 @@ impl Parser {
         };
         if self.cur_is(Colon) {
             self.lpop();
-            let body = self.try_reduce_block().map_err(|_| self.stack_dec())?;
+            let body = self.try_reduce_block().map_err(|_| {
+                self.stack_dec();
+                if let Some(err) = self.errs.last_mut() {
+                    err.set_hint("Lambda");
```

Comment:
When an error occurs the last element is available for extraction
Extract it and have a string as a hint